modules/programs/himalaya.nix

```diff
@@ -164,38 +154,45 @@ in {
   config = lib.mkIf himalaya.enable {
     home.packages = [ himalaya.package ];
 
-    xdg.configFile."himalaya/config.toml".source = let
-      enabledAccounts = lib.filterAttrs (_: account: account.himalaya.enable)
-        config.accounts.email.accounts;
-      accountsConfig = lib.mapAttrs mkAccountConfig enabledAccounts;
-      globalConfig = compactAttrs himalaya.settings;
-      allConfig = globalConfig // { accounts = accountsConfig; };
-    in tomlFormat.generate "himalaya-config.toml" allConfig;
-    systemd.user.services = let
-      inherit (config.services.himalaya-watch) enable environment settings;
-      optionalArg = key:
-        if (key ? settings && !isNull settings."${key}") then
-          [ "--${key} ${settings."${key}"}" ]
-        else
-          [ ];
-    in {
-      himalaya-watch = lib.mkIf enable {
+    xdg = {
+      configFile."himalaya/config.toml".source = let
+        enabledAccounts = lib.filterAttrs (_: account: account.himalaya.enable)
+          config.accounts.email.accounts;
+        accountsConfig = lib.mapAttrs mkAccountConfig enabledAccounts;
+        globalConfig = compactAttrs himalaya.settings;
+        allConfig = globalConfig // { accounts = accountsConfig; };
+      in tomlFormat.generate "himalaya-config.toml" allConfig;
+
+      desktopEntries.himalaya = lib.mkIf pkgs.stdenv.hostPlatform.isLinux {
+        type = "Application";
+        name = "himalaya";
+        genericName = "Email Client";
+        comment = "CLI to manage emails";
+        terminal = true;
+        exec = "himalaya %u";
+        categories = [ "Network" ];
+        mimeType = [ "x-scheme-handler/mailto" "message/rfc822" ];
+        settings = { Keywords = "email"; };
+      };
+    };
+
+    systemd.user.services."himalaya-watch@" =
```

Comment:
Unfortunately, HM doesn't play well with template units, in particular it makes it impossible to do automatic service start/stop/restart.

The more HM-esque solution would be to add an option `accounts.email.accounts.<name>.himalaya.enableWatch` and for each account with that enabled create a service "himalaya-watch-${account.name}".

Reply:
Ow, I see. Thank you for pointing it out!